contracts/contracts/levels/DoubleEntryPointFactory.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
+
+import './DoubleEntryPoint.sol';
+import './base/Level.sol';
+
+contract DoubleEntryPointFactory is Level {
+
+  function createInstance(address _player) override public payable returns (address) {
+    // Create legacy token
+    LegacyToken oldToken = new LegacyToken();
+    // Create a new Forta contract
+    Forta forta = new Forta();
+    // Create a new CryptoVault
+    CryptoVault vault = new CryptoVault(_player);
+    // Create latest token
+    DoubleEntryPoint newToken = new DoubleEntryPoint(address(oldToken), address(vault), address(forta), _player);
+    // Set underlying in CryptoVault
+    vault.setUnderlying(address(newToken));
+  
+    // Activate legacy support to newToken
+    oldToken.delegateToNewContract(DelegateERC20(address(newToken)));
+    
+    // Give CryptoVault some LGT (LegacyTokens)
+    oldToken.mint(address(vault), 100 ether);
+
+    return address(newToken);
+  }
+
+  function validateInstance(address payable _instance, address _player) override public returns (bool) {
+    DoubleEntryPoint instance = DoubleEntryPoint(_instance);
+    Forta forta = instance.forta();
+
+    // If user didn't set an agent, level failed.
+    address userAgent = address(forta.usersAgent(_player));
+    if(userAgent == address(0)) return false;
+
+    address vault = instance.cryptoVault();
+    CryptoVault cryptoVault = CryptoVault(vault);
+    try cryptoVault.sweepToken(IERC20(instance.delegatedFrom())) {
+      // If it didn't revert, it means no alerts have been raised by the agent. level failed.
+      return false;
+    } catch {
+      // If balance has been swept, level failed.
```

Comment:
One problem is that this would leave the instance unusable after one attempt to validate it.

There is a trick we can do to revert the effects of the sweep if the validation fails. We can define an external function along these lines:

```solidity
function __trySweep(cryptoVault, instance) external {
  try cryptoVault.sweepToken(IERC20(instance.delegatedFrom())) {
    revert abi.encode(false);
  } catch {
    revert abi.encode(instance.balanceOf(instance.cryptoVault()) > 0);
  }
}
```

The sweep is always reverted, and the returndata ("revertdata") encodes whether the sweep succeded.

Then use it like:

```solidity
(bool ok, bytes memory data) = this.__trySweep(...);
require(!ok);
bool swept = abi.decode(data, (bool));
...
```

Reply:
since `__trySweep` is inside the factory itself, that is not an external call, but I've implemented it returning `bool, bytes` as expected

Reply:
Ok got it.

Now the current implementation will:
- Revert if the sweep executes succesfully (reverting the state and leaving the instance untouched)
- Return the boolean of whether the cryptovault balance of underlying is zero after a failing call to sweep. The failing call to sweep should have avoided sweeping the balance so this boolean should be true if balance > 0, false otherwise. Those values are used by Ethernaut.sol to determine whether the level is solved or not.